newIDE/electron-app/app/LocalFilesystemWatcher.js

```diff
@@ -12,31 +12,38 @@ const getNewSubscriptionId = () => {
 const setupWatcher = (folderPath, fileWiseCallback, serializedOptions) => {
   const options = JSON.parse(serializedOptions);
   const newSubscriptionId = getNewSubscriptionId();
-  watcher
-    .subscribe(
-      folderPath,
-      (err, fileChangeEvents) => {
-        fileChangeEvents.forEach(fileChangeEvent =>
-          fileWiseCallback(fileChangeEvent.path)
-        );
+  const watcher = fileWatcher
+    .watch(folderPath, {
+      ignored: options.ignore,
+      ignoreInitial: true,
+      awaitWriteFinish: {
+        stabilityThreshold: 250,
+        pollInterval: 100,
       },
-      options
+    })
+    .on(
+      'change',
+      // TODO: Is it safe to let it like that since the OS could for some reason
+      // do never-ending operations on the folder or its children, making the debounce
+      // never ending.
+      fileWiseCallback
     )
-    .then(subscription => {
-      subscriptions[newSubscriptionId] = subscription;
-    });
+    .on('unlink', fileWiseCallback)
+    .on('add', fileWiseCallback);
+
+  subscriptionCancelers[newSubscriptionId] = () => watcher.unwatch(folderPath);
```

Comment:
just to check, this function is not async?

Reply:
Nope!